### ISSUE TITLE
Error: error reading IdentityGroup

### ISSUE THREAD

**Issue by tristanmorgan** [hashicorp/terraform-provider-vault#1269]
### Terraform Version
```
$ terraform -version
Terraform v1.1.2
on darwin_arm64
+ provider registry.terraform.io/hashicorp/vault v3.1.0
```

### Affected Resource(s)
- vault_identity_group

### Terraform Configuration Files
```hcl
data "vault_policy_document" "admin" {
  rule {
    path         = "*"
    capabilities = ["create", "read", "update", "delete", "list", "sudo"]
  }
}

resource "vault_policy" "admin" {
  name   = "admin"
  policy = data.vault_policy_document.admin.hcl
}

resource "vault_github_auth_backend" "github" {
  path           = "github"
  organization   = "company"
  description    = "Authenticate using GitHub"
  token_policies = ["default"]

  tune {
    default_lease_ttl = "1h"
    max_lease_ttl     = "2h"
    token_type        = "default-service"
  }
}

resource "vault_identity_group" "company_engineers" {
  name     = "company-engineers"
  type     = "external"
  policies = [vault_policy.admin.name]
}

resource "vault_identity_group_alias" "company_engineers" {
  name           = "company-engineers"
  mount_accessor = vault_github_auth_backend.github.accessor
  canonical_id   = vault_identity_group.company_engineers.id
}
```
The issue doesn't seem to affect vault running in dev so maybe timing of audit logs and raft are an effect.

### Debug Output
[terraform debug output](https://gist.github.com/tristanmorgan/e70d689e13b95bd331fada678c2376f8)

### Expected Behavior
Version 3.0.1 of the provider did not produce an error, this was expected.

### Actual Behavior
Using 3.1.0 (latest as of 2021-12-22) looking at the debug output and vault logs, it seems the vault token is *not* being passed in the api request, producing the error.

### Steps to Reproduce
1. `terraform init`
2. `terraform apply`

### Important Factoids
Vault was running with Raft integrated storage with audit logs enabled to write to files. vault dev mode (in-mem) didn't produce the errors.

### References
it may be related to
- GH-1263

**Comment by neurostream** [comment]
Started receiving this in my build chain 4 hours and 40 minutes ago.

Confirmed that if I read this this via the Vault CLI ( `vault read /identity/group/id/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee` that it is readable )

**Comment by benashz** [comment]
@tristanmorgan thank you for reporting this issue. We should have a fix out today.

**Comment by benashz** [comment]
Released in [v3.1.1](https://github.com/hashicorp/terraform-provider-vault/releases/tag/v3.1.1)

**Comment by neurostream** [comment]
> Released in [v3.1.1](https://github.com/hashicorp/terraform-provider-vault/releases/tag/v3.1.1)

This _Set parent client's token on its clone_ adjustment definitely cleared up this IdentityGroup read issue for me as well.  Good stuff!
